src/main/scala/org/constellation/infrastructure/endpoints/SnapshotEndpoints.scala

```diff
@@ -116,18 +119,33 @@ class SnapshotEndpoints[F[_]](implicit F: Concurrent[F], C: ContextShift[F]) ext
         redownloadService.getAcceptedSnapshots().map(_.asJson).flatMap(Ok(_))
     }
 
-  private def getPeerProposals(nodeId: Id, redownloadService: RedownloadService[F]): HttpRoutes[F] =
+  private def getPeerProposals(selfId: Id, redownloadService: RedownloadService[F]): HttpRoutes[F] =
     HttpRoutes.of[F] {
       case GET -> Root / "peer" / peerId / "snapshot" / "created" =>
         val peerProposals =
-          if (Id(peerId) == nodeId)
+          if (Id(peerId) == selfId)
             redownloadService.getCreatedSnapshots()
           else
-            redownloadService.getPeerProposals(nodeId).map(_.getOrElse(Map.empty))
+            redownloadService.getPeerProposals(Id(peerId)).map(_.getOrElse(Map.empty))
 
         peerProposals.map(_.asJson).flatMap(Ok(_))
     }
 
+  private def queryPeerProposals(selfId: Id, redownloadService: RedownloadService[F]): HttpRoutes[F] =
+    HttpRoutes.of[F] {
+      case req @ POST -> Root / "snapshot" / "proposal" / "query" =>
```

Comment:
wdyt about using `_query` ending name?
Right now it looks like this endpoint will create a query for snapshot proposal, or sth like that ;)

We use(d) `_batch`, so we can use `_query` as well, to mark it as a non-common thing.

Reply:
I actually thought about something similar. I will change it.